### PR TITLE
Custom user-defined ability+skill rolls from drop-down menus, drop-down for specialties

### PR DESCRIPTION
**Pull request by Veichassur**
I'm really missing some functionality from roll20, like user-defined requent rolls that have their own name and dice pool.
So far it's only Ability+Skill combo, without any modifiers, might need to extend it a little.
Also very hacky, I have absolutely no idea what I'm doing, I'm not a JS programmer.

### CODE REVIEW

**Comment by Veichassur** [comment]
image

**Comment by Rayji96** [comment]
Isn't it in the rules that a specialty in a skill would apply to any attribute roll using that skill? I could be wrong

**Comment by Rayji96** [comment]
On an unrelated note, thank you for posting screenshots! Really helps me quickly understand the PR :)

**Comment by Veichassur** [comment]
You're right! It's still very much WIP, but my plan for the speciaties is
that when you click on them it would show the general roll dialog with the
related skill and +1 modifier already set up for it.

The customized rolls are a separate feature for frequently used rolls, like
memory, perception or your character's predator roll. My bad, I just
copy&pasted the specialty section, but this will have a different header.

Also, imho this would be another prime example where you don't want to edit
the roll's dice pool (ability+skill combo) on a separate item page but
rather directly on the character sheet.

**Comment by Veichassur** [comment]
image
image

**Comment by Veichassur** [comment]
Would need a nice .css update for the selections, I'm just not capable of doing that. :D
Custom dice pools need to be changeable on the character sheet because of UX reasons, I know they break data encapsulation but otherwise items wouldn't know about the exact list of attributes and skills. (I wouldn't copy&paste those everywhere)
Custom dice pools might not even need to be items, it was just easier this way because they can also be added/deleted to/from the sheet.

File: lang/es.json [hunk 1]
```diff
@@ -76,6 +77,7 @@
     "VTM5E.AddItem": "Añadir",
     "VTM5E.Edit": "Editar",
     "VTM5E.Delete": "Borrar",
+    "VTM5E.CustomRolls": "Custom Rolls",
```
Comment (Rayji96) [inline lang/es.json:80]: Can you find the right translation or leave it as VTM5E.CustomRolls? Makes it hard for me to check for missing localizations.

File: lang/fr.json [hunk 2]
```diff
@@ -76,6 +77,7 @@
     "VTM5E.AddItem": "Ajouter",
     "VTM5E.Edit": "Éditer",
     "VTM5E.Delete": "Supprimer",
+    "VTM5E.CustomRolls": "Custom Rolls",
```
Comment (Rayji96) [inline lang/fr.json:80]: ditto

File: lang/pt-BR.json [hunk 3]
```diff
@@ -76,6 +77,7 @@
     "VTM5E.AddItem": "Adicionar",
     "VTM5E.Edit": "Editar",
     "VTM5E.Delete": "Apagar",
+    "VTM5E.CustomRolls": "Custom Rolls",
```
Comment (Rayji96) [inline lang/pt-BR.json:80]: ditto

**Comment by tom959** [comment]
could you change the de.json so i can translate it, to match your additions?

**Comment by Veichassur** [comment]
Something like this is okay for everyone? I think it's better if the keys are already there so the jsons keep their structure, just the non-translated elements have a special identifier in them.

**Comment by tom959** [comment]
That should be good, thank you

**Review by Rayji96 — APPROVED** [review]

**Comment by Rayji96** [comment]
Custom rolls are such a good idea thank you :)

**Comment by Rayji96** [comment]
Specialties are forever tied to a specific skill, aren't they? I'm going to remove the dropdown from specialty and move it into the actual item sheet itself

**Pull request merged**